dz13/src/app.js

```diff
@@ -0,0 +1,126 @@
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+
+// function setKeysAndObjectInfo(user) {
+//   const entriesUser = Object.entries(user)
+//   const keyUser = []
+//   for (const key of entriesUser) {
+//     key.pop()
+//     keyUser.push(key)
+//   }
+//   console.log(keyUser.flat())
+// }
+function setKeysAndObjectInfo(user, type) {
+  if (type === 'key') {
+    const newUser = Object.keys(user)
+    return newUser
+  }
+  const newUser = Object.values(user)
+  return newUser
+}
+
+function logInCheck(cart) {
+  const newCart = Object.entries(cart)
+  for (const logCart of newCart) {
+    console.log(`${logCart[0]}: ${logCart[1]} шт.`)
+  }
+}
+
+function mergeSettings(defaultSettings, userPreferences) {
+  const resultSettings = { ...defaultSettings, ...userPreferences }
+  return resultSettings
+}
+
+function checkUserAge(user) {
+  try {
+    if (user.age < 18) {
+      throw new Error('Меньше 18 лет')
+    }
+    console.log('Больше 18 лет(')
+  } catch (error) {
+    if (error.message === 'Меньше 18 лет') {
+      console.log('Меньше 18 лет(')
+    }
+  }
+}
+
+function checkNumber(number) {
+  return new Promise((resolve, reject) => {
+    if (number > 10) {
+      resolve('Число больше 10')
+    } else {
+      reject('Число меньше или равно 10')
```

Comment:
```suggestion
      reject(new Error('Число меньше или равно 10'))
```

в reject лучше выкидвать не просто строку, а ошибку с таким сообщением (по идее eslint у тебя на это указывал)